libtrap/src/ifc_file.c

```diff
@@ -150,9 +177,8 @@ char *create_filename_from_time(void *priv)
       return NULL;
    }
 
-   strncpy(new_filename, config->filename, config->filename_base_length);
-   
-   strftime(new_filename + config->filename_base_length, 14, ".%Y%m%d%H%M", localtime(&config->starting_time));
+   strftime(new_filename, config->filename_base_length + 6, config->filename, tm_info);
+   strftime(new_filename + (strlen(new_filename)), 14, ".%Y%m%d%H%M", localtime(&config->starting_time));
```

Comment:
I know in previous version the return value of strftime() was not checked, however you should do it now, since the formating string is defined by the user.

Reply:
Why don't we use a single call instead of 2?